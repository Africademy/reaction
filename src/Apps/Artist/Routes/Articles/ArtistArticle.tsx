import { Sans, Serif } from "@artsy/palette"
import React, { SFC } from "react"
import { Box } from "Styleguide/Elements/Box"
import { Flex } from "Styleguide/Elements/Flex"
import { Col, Row } from "Styleguide/Elements/Grid"
import { Image } from "Styleguide/Elements/Image"
import { Separator } from "Styleguide/Elements/Separator"
import { Spacer } from "Styleguide/Elements/Spacer"
import { Responsive } from "Utils/Responsive"

interface ArticleItemProps {
  imageUrl: string
  date: string
  author: string
  title: string
  href: string
}

export const ArticleItem: SFC<ArticleItemProps> = props => {
  return (
    <Responsive>
      {({ xs }) => {
        if (xs) return <SmallArticleItem {...props} />
        else return <LargeArticleItem {...props} />
      }}
    </Responsive>
  )
}

const LargeArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title } = props

  return (
    <React.Fragment>
      <Row>
        <Col sm={2}>
          <Date size="3" weight="medium">
            {date}
          </Date>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={12}>
              <Title size="5">
                <a href={href} className="noUnderline">
                  {title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {author}
              </Credit>
            </Box>
          </Flex>
        </Col>
        <Col sm={2}>
          <Image width="135px" height="85px" src={imageUrl} />
        </Col>
      </Row>

      {/* FIXME: Weird block height issue... */}
      <Spacer mt={2} />
      <Separator />
      <Spacer mt={3} />
    </React.Fragment>
  )
}

const SmallArticleItem: SFC<ArticleItemProps> = props => {
  const { author, date, href, imageUrl, title } = props

  return (
    <React.Fragment>
      <Flex justifyContent="space-between">
        <Box pr={3}>
          <Date size="1" weight="medium">
            {date}
          </Date>
          <Spacer mb={0.5} />
          <Title size="2">
            <a href={href} className="noUnderline">
              {title}
            </a>
          </Title>
          <Spacer mb={0.5} />
          <Credit size="1" color="black60">
            {author}
          </Credit>
        </Box>
        <Image width="80px" height="50px" src={imageUrl} />
      </Flex>

      {/* FIXME: Weird block height issue... */}
      <Spacer mt={2} />
      <Separator />
    </React.Fragment>
  )
}

const Date = Sans
const Title = Serif
const Credit = Serif
