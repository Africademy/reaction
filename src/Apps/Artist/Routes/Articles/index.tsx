import { Articles_artist } from "__generated__/Articles_artist.graphql"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"
import { ArticlesRefetchContainer as Articles } from "./ArticlesRefetchContainer"

export interface ArticlesRouteProps {
  artist: Articles_artist
}

export const ArticlesRoute: React.SFC<ArticlesRouteProps> = props => {
  return <Articles artist={props.artist as any} />
}

export const ArticlesRouteFragmentContainer = createFragmentContainer(
  ArticlesRoute,
  graphql`
    fragment Articles_artist on Artist {
      ...ArticlesRefetchContainer_artist
    }
  `
)
