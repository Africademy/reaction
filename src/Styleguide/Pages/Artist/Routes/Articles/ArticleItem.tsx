import { Sans, Serif } from "@artsy/palette"
import React from "react"
import { Box } from "Styleguide/Elements/Box"
import { Flex } from "Styleguide/Elements/Flex"
import { Col, Row } from "Styleguide/Elements/Grid"
import { Image } from "Styleguide/Elements/Image"
import { Responsive } from "Styleguide/Utils/Responsive"

interface ArticleItemProps {
  imageUrl: string
  date: string
  author: string
  title: string
  href: string
}

export const ArticleItem = (props: ArticleItemProps) => {
  return (
    <Responsive>
      {({ xs }) => {
        if (xs) return <SmallArticleItem {...props} />
        else return <LargeArticleItem {...props} />
      }}
    </Responsive>
  )
}

const LargeArticleItem = (props: ArticleItemProps) => {
  return (
    <React.Fragment>
      <Row>
        <Col sm={2}>
          <Date size="3" weight="medium">
            {props.date}
          </Date>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={12}>
              <Title size="5">
                <a href={props.href} className="noUnderline">
                  {props.title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {props.author}
              </Credit>
            </Box>
          </Flex>
        </Col>
        <Col sm={2}>
          <Image width="135px" height="85px" src={props.imageUrl} />
        </Col>
      </Row>
    </React.Fragment>
  )
}

const SmallArticleItem = (props: ArticleItemProps) => {
  return (
    <React.Fragment>
      <Row>
        <Col sm={2}>
          <Box mb={0.5}>
            <Date size="1" weight="medium">
              {props.date}
            </Date>
          </Box>
        </Col>
        <Col sm={8}>
          <Flex>
            <Box pr={3}>
              <Title size="2">
                <a href={props.href} className="noUnderline">
                  {props.title}
                </a>
              </Title>
              <Credit size="2" color="black60">
                {props.author}
              </Credit>
            </Box>
            <Image width="80px" height="50px" src={props.imageUrl} />
          </Flex>
        </Col>
      </Row>
    </React.Fragment>
  )
}

const Date = Sans
const Title = Serif
const Credit = Serif
