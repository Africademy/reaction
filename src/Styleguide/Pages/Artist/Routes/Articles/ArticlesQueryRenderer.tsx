import { ContextConsumer, ContextProps } from "Components/Artsy"
import React from "react"
import { graphql, QueryRenderer } from "react-relay"
import { ArticlesRefetchContainer } from "./ArticlesRefetchContainer"

interface Props extends ContextProps {
  artistID: string
}

const PAGE_SIZE = 10

export const ArticlesQueryRenderer = ContextConsumer(
  class extends React.Component<Props> {
    render() {
      const { artistID, relayEnvironment } = this.props
      return (
        <QueryRenderer
          environment={relayEnvironment}
          query={graphql`
            query ArticlesQueryRendererQuery($artistID: String!, $first: Int!) {
              artist(id: $artistID) {
                ...ArticlesRefetchContainer_artist @arguments(first: $first)
              }
            }
          `}
          variables={{ artistID, first: PAGE_SIZE }}
          render={({ props }) => {
            if (props) {
              return <ArticlesRefetchContainer artist={props.artist} />
            } else {
              return null
            }
          }}
        />
      )
    }
  }
)
