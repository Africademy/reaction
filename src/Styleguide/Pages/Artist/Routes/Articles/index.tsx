import React from "react"
import { ArticlesRefetchContainer } from "./ArticlesRefetchContainer"

interface Props {
  artist: any
}

export class ArticlesRoute extends React.Component<Props> {
  render() {
    return <ArticlesRefetchContainer artist={this.props.artist} />
  }
}
