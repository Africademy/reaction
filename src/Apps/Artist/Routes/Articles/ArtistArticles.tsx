import { ArtistArticles_artist } from "__generated__/ArtistArticles_artist.graphql"
import React, { Component } from "react"
import { createRefetchContainer, graphql, RelayRefetchProp } from "react-relay"
import { PaginationFragmentContainer as Pagination } from "Styleguide/Components/Pagination"
import { Flex } from "Styleguide/Elements/Flex"
import { Col, Row } from "Styleguide/Elements/Grid"
import { ArticleItem } from "./ArtistArticle"

import {
  LoadingArea,
  LoadingAreaState,
} from "Apps/Artist/Components/LoadingArea"

const PAGE_SIZE = 10

interface ArtistArticlesProps {
  relay: RelayRefetchProp
  artist: ArtistArticles_artist
}

export class ArtistArticles extends Component<
  ArtistArticlesProps,
  LoadingAreaState
> {
  state = {
    isLoading: false,
  }

  loadNext = () => {
    const {
      artist: {
        articlesConnection: {
          pageInfo: { hasNextPage, endCursor },
        },
      },
    } = this.props

    if (hasNextPage) {
      this.loadAfter(endCursor)
    }
  }

  loadAfter = cursor => {
    this.toggleLoading(true)

    this.props.relay.refetch(
      {
        first: PAGE_SIZE,
        after: cursor,
        artistID: this.props.artist.id,
        before: null,
        last: null,
      },
      null,
      error => {
        this.toggleLoading(false)

        if (error) {
          console.error(error)
        }
      }
    )
  }

  toggleLoading = isLoading => {
    this.setState({
      isLoading,
    })
  }

  render() {
    return (
      <React.Fragment>
        <span id="jumpto--artistArticles" />

        <Row>
          <Col>
            <LoadingArea isLoading={this.state.isLoading}>
              {this.props.artist.articlesConnection.edges.map(
                ({ node }, index) => {
                  return (
                    <ArticleItem
                      title={node.thumbnail_title}
                      imageUrl={node.thumbnail_image.resized.url}
                      date={node.published_at}
                      author={node.author.name}
                      href={node.href}
                      key={index}
                    />
                  )
                }
              )}
            </LoadingArea>
          </Col>
        </Row>
        <Row>
          <Col>
            <Flex mb={2} justifyContent="flex-end">
              <Pagination
                pageCursors={
                  this.props.artist.articlesConnection.pageCursors as any
                }
                onClick={this.loadAfter}
                onNext={this.loadNext}
                scrollTo="#jumpto-RouteTabs"
              />
            </Flex>
          </Col>
        </Row>
      </React.Fragment>
    )
  }
}

export const ArtistArticlesRefetchContainer = createRefetchContainer(
  ArtistArticles,
  {
    artist: graphql`
      fragment ArtistArticles_artist on Artist
        @argumentDefinitions(
          first: { type: "Int", defaultValue: 10 }
          last: { type: "Int" }
          after: { type: "String" }
          before: { type: "String" }
        ) {
        id
        articlesConnection(
          first: $first
          after: $after
          before: $before
          last: $last
          sort: PUBLISHED_AT_DESC
          in_editorial_feed: true
        ) {
          pageInfo {
            hasNextPage
            endCursor
          }
          pageCursors {
            ...Pagination_pageCursors
          }
          edges {
            node {
              href
              thumbnail_title
              author {
                name
              }
              published_at(format: "MMM d, YYYY")
              thumbnail_image {
                resized(width: 300) {
                  url
                }
              }
              href
            }
          }
        }
      }
    `,
  },
  graphql`
    query ArtistArticlesQuery(
      $first: Int
      $last: Int
      $after: String
      $before: String
      $artistID: String!
    ) {
      artist(id: $artistID) {
        ...ArtistArticles_artist
          @arguments(first: $first, last: $last, after: $after, before: $before)
      }
    }
  `
)
