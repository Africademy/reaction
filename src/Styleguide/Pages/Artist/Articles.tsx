import React from "react"
import {
  graphql,
  QueryRenderer,
  createRefetchContainer,
  RelayRefetchProp,
} from "react-relay"
import { ContextProps, ContextConsumer } from "Components/Artsy"
import styled from "styled-components"
import { Col, Row } from "Styleguide/Elements/Grid"
import { Box } from "Styleguide/Elements/Box"
import { Flex } from "Styleguide/Elements/Flex"
import { Separator } from "Styleguide/Elements/Separator"
import { ArticleItem } from "./ArticleItem"
import { Pagination } from "Styleguide/Components/Pagination"

import { Articles_artist } from "__generated__/Articles_artist.graphql"

interface ArticlesProps {
  relay: RelayRefetchProp
  artist: Articles_artist
}

const PAGE_SIZE = 10

// Renders list of articles, we'll use this in our refetch container.
class ArticlesContainer extends React.Component<ArticlesProps> {
  loadPrev = () => {
    const {
      startCursor,
      hasPreviousPage,
    } = this.props.artist.articlesConnection.pageInfo
    if (hasPreviousPage) {
      this.loadBefore(startCursor)
    }
  }

  loadNext = () => {
    const {
      hasNextPage,
      endCursor,
    } = this.props.artist.articlesConnection.pageInfo
    if (hasNextPage) {
      this.loadAfter(endCursor)
    }
  }

  loadBefore(cursor) {
    this.props.relay.refetch(
      {
        first: null,
        before: cursor,
        artistID: this.props.artist.id,
        after: null,
        last: PAGE_SIZE,
      },
      null,
      error => {
        if (error) {
          console.error(error)
        }
      }
    )
  }

  loadAfter = cursor => {
    this.props.relay.refetch(
      {
        first: PAGE_SIZE,
        after: cursor,
        artistID: this.props.artist.id,
        before: null,
        last: null,
      },
      null,
      error => {
        if (error) {
          console.error(error)
        }
      }
    )
  }

  renderPagination() {
    return (
      <div>
        <Pagination
          {...this.props.artist.articlesConnection.pageCursors}
          onClick={this.loadAfter}
          onNext={this.loadNext}
          onPrev={this.loadPrev}
        />
      </div>
    )
  }

  render() {
    return (
      <div>
        <Row>
          <Col>
            <ArticleList>
              {this.props.artist.articlesConnection.edges.map(({ node }) => {
                return (
                  <ArticleItem
                    title={node.thumbnail_title}
                    imageUrl={node.thumbnail_image.resized.url}
                    date={node.published_at}
                    author={node.author.name}
                    href={node.href}
                  />
                )
              })}
            </ArticleList>
          </Col>
        </Row>

        <Box my={2}>
          <Separator />
        </Box>

        <Row>
          <Col>
            <Flex mb={2} justifyContent="flex-end">
              {this.renderPagination()}
            </Flex>
          </Col>
        </Row>
      </div>
    )
  }
}

// This is the actual Refetch Container we want to use.
const ArticlesRefetchContainer = createRefetchContainer(
  ArticlesContainer,
  {
    artist: graphql`
      fragment Articles_artist on Artist
        @argumentDefinitions(
          first: { type: "Int" }
          last: { type: "Int" }
          after: { type: "String" }
          before: { type: "String" }
        ) {
        id
        articlesConnection(
          first: $first
          after: $after
          before: $before
          last: $last
          sort: PUBLISHED_AT_DESC
          in_editorial_feed: true
        ) {
          pageInfo {
            hasNextPage
            hasPreviousPage
            startCursor
            endCursor
          }
          pageCursors {
            around {
              cursor
              page
              isCurrent
            }
            first {
              cursor
              page
              isCurrent
            }
            last {
              cursor
              page
              isCurrent
            }
          }
          edges {
            node {
              href
              thumbnail_title
              author {
                name
              }
              published_at(format: "MMM d, YYYY")
              thumbnail_image {
                resized(width: 300) {
                  url
                }
              }
              href
            }
          }
        }
      }
    `,
  },
  graphql`
    query ArticlesQuery(
      $first: Int
      $last: Int
      $after: String
      $before: String
      $artistID: String!
    ) {
      artist(id: $artistID) {
        ...Articles_artist
          @arguments(first: $first, last: $last, after: $after, before: $before)
      }
    }
  `
)

interface Props extends ContextProps {
  artistID: string
}

class ArticlesQueryRenderer extends React.Component<Props> {
  render() {
    const { artistID, relayEnvironment } = this.props
    return (
      <QueryRenderer
        environment={relayEnvironment}
        query={graphql`
          query ArticlesIndexQuery($artistID: String!, $first: Int!) {
            artist(id: $artistID) {
              ...Articles_artist @arguments(first: $first)
            }
          }
        `}
        variables={{ artistID, first: PAGE_SIZE }}
        render={({ props }) => {
          if (props) {
            return <ArticlesRefetchContainer artist={props.artist} />
          } else {
            return null
          }
        }}
      />
    )
  }
}

export const ArticlesContent = ContextConsumer(ArticlesQueryRenderer)

const ArticleList = styled.div``
