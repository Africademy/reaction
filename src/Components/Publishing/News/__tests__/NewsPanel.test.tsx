import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"

import { FeatureArticle, NewsArticle } from "../../Fixtures/Articles"
import { NewsPanel } from "../NewsPanel"

it("Renders a news panel properly", () => {
  const component = renderer
    .create(<NewsPanel articles={[NewsArticle]} />)
    .toJSON()
  expect(component).toMatchSnapshot()
})

it("Renders article headlines", () => {
  const wrapper = mount(<NewsPanel articles={[NewsArticle, FeatureArticle]} />)
  expect(wrapper.text()).toMatch(NewsArticle.title)
  expect(wrapper.text()).toMatch(FeatureArticle.title)
})
