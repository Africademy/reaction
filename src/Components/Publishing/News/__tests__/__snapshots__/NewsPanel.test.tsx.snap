// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a news panel properly 1`] = `
.c0 {
  padding: 20px;
  border: 1px solid #e5e5e5;
  border-radius: 3px;
}

.c0 a {
  color: black;
}

.c0 a:hover {
  color: #999999;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.c3 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
}

.c2 {
  font-family: Unica77LLWebMedium , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 19px;
  line-height: 1.5em;
  position: absolute;
  left: 0;
}

.c4 {
  padding: 10px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 18px;
  line-height: 1.4em;
  padding: 10px 0;
  cursor: pointer;
  text-decoration: none;
}

.c6 {
  font-family: Unica77LLWebRegular , 'Arial', 'serif';
  -webkit-font-smoothing: antialiased;
  font-size: 16px;
  line-height: 1.1em;
}

@media (max-width:720px) {
  .c3 {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }
}

@media (max-width:720px) {
  .c2 {
    font-family: Unica77LLWebMedium , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }
}

@media (max-width:720px) {
  .c4 {
    padding: 0 0 10px;
  }
}

@media (max-width:720px) {
  .c5 {
    font-family: 'Adobe Garamond W08', 'adobe-garamond-pro', 'AGaramondPro-Regular', 'Times New Roman', 'Times', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 15px;
    line-height: 1.25em;
    line-height: 1.4em;
  }
}

@media (max-width:720px) {
  .c6 {
    font-family: Unica77LLWebRegular , 'Arial', 'serif';
    -webkit-font-smoothing: antialiased;
    font-size: 12px;
    line-height: 1.4em;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      News
    </div>
    <div
      className="c3"
    >
      Apr 5
    </div>
  </div>
  <div
    className="c4"
  >
    <a
      className="c5"
      href={undefined}
    >
      The oldest known depiction of a supernova was found in a work of 5,000 year old rock art, scientists believe.
    </a>
  </div>
  <div
    className="c6"
  >
    <a
      href="/news"
    >
      More In News
    </a>
  </div>
</div>
`;
